components/brave_new_tab_ui/reducers/new_tab_reducer.tsx

```diff
@@ -358,6 +358,24 @@ export const newTabReducer: Reducer<NewTab.State | undefined> = (state: NewTab.S
       state.rewardsState.grants = updatedGrants
       break
 
+    case types.ON_GRANT_FINISH:
+      const properties = payload.properties
+
+      if (properties.status !== 0) {
+        break
+      }
+
+      state = { ...state }
+      const oldNotifications = state.rewardsState.dismissedNotifications
+
+      oldNotifications.push(payload.id)
+      state.rewardsState.dismissedNotifications = oldNotifications
+
+      state.rewardsState.grants = state.rewardsState.grants.filter((grant) => {
```

Comment:
why don't we just fetch new grants after grant is finished? This was current grant that was in the list before will be automatically removed

Reply:
We could do that, but we have to manage notifications separately, because if we have a case like, user dismisses grant notification via new tab widget, but doesn't claim the grant, then refreshes the page - if we are just mimicking perfectly what is in list of grants the notification will reappear

Reply:
but doesn't this happens anyway? `case types.ON_GRANT` you always add them back or am I missing something?